src/fsharp/CompileOps.fs

```diff
@@ -4879,27 +4886,34 @@ let ProcessMetaCommandsFromInput
                List.fold (fun state d -> nowarnF state (m,d)) state numbers
 
             | ParsedHashDirective(("reference" | "r"),args,m) -> 
-               if not canHaveScriptMetaCommands then
-                   errorR(HashReferenceNotAllowedInNonScript m)
+                if not canHaveScriptMetaCommands then
+                    errorR(HashReferenceNotAllowedInNonScript m)
+
+                let reportError errorType error =
+                    match errorType with
+                    | ErrorReportType.Warning -> warning(Error(error,m))
+                    | ErrorReportType.Error -> errorR(Error(error, m))
+
+                match args with
+                | [path] ->
+                    matchedm <- m
+                    let dm = tcConfig.dependencyProvider.TryFindDependencyManagerInPath(tcConfig.compilerToolPaths, tcConfig.outputDir |> Option.defaultValue "" , reportError, path)
```

Comment:
`tcConfig.outputDir |> Option.defaultValue ""` -> might be better just to bind to a value above for debuggability? That way it's easier to run under the debugger and see why it might be `""`

Reply:
Looks like you do that elsewhere in the code

Reply:
@cartermp the ````let dm```` itself is an example of me doing hoisting expressions for the sake of debuggability so, I can offer no plausible rebuttal.